### PR TITLE
Added material editor+creator to UI

### PR DESCRIPTION
**Pull request by jmjaffe37**
title

### CODE REVIEW

File: SHOE/Source/EditingUI.cpp [hunk 1]
```diff
@@ -972,6 +987,228 @@ void EditingUI::GenerateEditingUI() {
...
+
+		ImGui::Text(node.c_str());
+
+		if (ImGui::ArrowButton("Previous Material", ImGuiDir_Left)) {
+			materialUIIndex--;
+			if (materialUIIndex < 0) {
+				materialUIIndex = globalAssets.GetMaterialArraySize() - 1;
+			}
+			this->materialUIIndex = materialUIIndex;
+			//globalAssets.currentSky = globalAssets.GetSkyAtID(materialUIIndex);
```
Comment (crigney3) [inline SHOE/Source/EditingUI.cpp:1008]: Please remove these commented out lines - you were likely using them for reference, which is great, but now they just add clutter.

File: SHOE/Source/EditingUI.cpp [hunk 2]
```diff
@@ -972,6 +987,228 @@ void EditingUI::GenerateEditingUI() {
...
+		float currRefractionScale = currentMaterial->GetRefractionScale();
+		ImGui::DragFloat("Refraction Scale ", &currRefractionScale);
+		currentMaterial->SetRefractionScale(currRefractionScale);
+
+		ImGui::Separator();
+		std::shared_ptr<Texture> currentTexture = currentMaterial->GetTexture();
+		std::shared_ptr<Texture> currentNormalMap = currentMaterial->GetNormalMap();
+		std::shared_ptr<Texture> currentMetalMap = currentMaterial->GetMetalMap();
+		std::shared_ptr<Texture> currentRoughMap = currentMaterial->GetRoughMap();
+		//std::shared_ptr<Texture> currentTexture = globalAssets.GetTextureAtID(textureUIIndex);
```
Comment (crigney3) [inline SHOE/Source/EditingUI.cpp:1058]: Same here

File: SHOE/Source/EditingUI.cpp [hunk 3]
```diff
@@ -972,6 +987,228 @@ void EditingUI::GenerateEditingUI() {
...
+		};
+		ImGui::SameLine();
+
+		if (ImGui::ArrowButton("Next Paterial", ImGuiDir_Right)) {
+			materialUIIndex++;
+			if (materialUIIndex > globalAssets.GetMaterialArraySize() - 1) {
+				materialUIIndex = 0;
+			}
+			this->materialUIIndex = materialUIIndex;
+			//globalAssets.currentSky = globalAssets.GetSkyAtID(materialUIIndex);
```
Comment (crigney3) [inline SHOE/Source/EditingUI.cpp:1018]: Same here

File: SHOE/Source/EditingUI.cpp (hunk 1 above)
Comment (jmjaffe37) [reply SHOE/Source/EditingUI.cpp]: Resolved in force pushed commit

File: SHOE/Source/EditingUI.cpp (hunk 2 above)
Comment (jmjaffe37) [reply SHOE/Source/EditingUI.cpp]: Resolved in force push

File: SHOE/Source/EditingUI.cpp (hunk 3 above)
Comment (jmjaffe37) [reply SHOE/Source/EditingUI.cpp]: Resolved in force push

**Pull request merged**